### PR TITLE
fix: delete assert self.debtRatio <= MAX_BPS

### PR DESCRIPTION
**Pull request by zgfzgf**
update self debt Ratio usage
This is a better way to control assets.
Solve a small part of this issue #376 

origin:
vault_debtLimit: uint256 = self.debtRatio * vault_totalAssets / MAX_BPS 
If this control is needed. we should add other var control it.

### CODE REVIEW

**Comment by steffenix** [comment]
We need to find a solution to this one https://github.com/yearn/yearn-vaults/pull/375 before going further on changing code related to debtRatio.

File: contracts/Vault.vy [hunk 1]
```diff
@@ -1181,7 +1181,6 @@ def addStrategy(
     assert self.token.address == Strategy(strategy).want()
 
     # Check strategy parameters
-    assert self.debtRatio + debtRatio <= MAX_BPS
```
Comment (fubuloubu) [inline contracts/Vault.vy:1184]: Not sure why you'd want to delete this, as it should ensure that `self.debtRatio <= 100%`, which is a necessary invariant for this code to work correctly

File: contracts/Vault.vy [hunk 2]
```diff
@@ -1225,7 +1224,6 @@ def updateStrategyDebtRatio(
     self.debtRatio -= self.strategies[strategy].debtRatio
     self.strategies[strategy].debtRatio = debtRatio
     self.debtRatio += debtRatio
-    assert self.debtRatio <= MAX_BPS
```
Comment (fubuloubu) [inline contracts/Vault.vy:1228]: Similar to above

File: contracts/Vault.vy [hunk 3]
```diff
@@ -1472,10 +1470,12 @@ def _creditAvailable(strategy: address) -> uint256:
     # See note on `creditAvailable()`.
     if self.emergencyShutdown:
         return 0
+    if self.debtRatio == 0:
```
Comment (fubuloubu) [inline contracts/Vault.vy:1473]: Could probably do this
```suggestion
    if self.debtRatio == 0 or self.strategies[strategy].debtRatio == 0:
```

But add caching for those two values e.g. `strategy_debtRatio` and `debtRatio` memory variables

File: contracts/Vault.vy (hunk 1 above)
Comment (zgfzgf) [reply contracts/Vault.vy:1184]: update code. It is OK
https://github.com/yearn/yearn-vaults/blob/f5a576b8c9ae07130df28abedadcad90e49f93f0/contracts/Vault.vy#L1478-L1480
vault_debtLimit: uint256 =  vault_totalAssets
    vault_totalDebt: uint256 = self.totalDebt
    strategy_debtLimit: uint256 = self.strategies[strategy].debtRatio * vault_totalAssets / self.debtRatio

File: contracts/Vault.vy (hunk 3 above)
Comment (zgfzgf) [reply contracts/Vault.vy:1473]: yes

**Comment by zgfzgf** [comment]
core: self.debtRatio can't control asset. 
https://github.com/yearn/yearn-vaults/blob/f5a576b8c9ae07130df28abedadcad90e49f93f0/contracts/Vault.vy#L1478

**Pull request closed without merging**